File: day_two/part_two.py
import shlex


class Line:
    def __init__(self, raw_line: str):
        self.raw_line = raw_line
        self.shlex_parts()

    def shlex_parts(self):
        num_range, letter, self.password = shlex.split(self.raw_line)
        self.first_num, self.second_num = [int(num) for num in num_range.split("-")]
        self.letter = letter[0]
        print(raw_line, [self.first_num, self.second_num, self.letter, self.password])

    def check(self) -> bool:
        first = self.password[self.first_num-1] == self.letter
        second = self.password[self.second_num-1] == self.letter
        if first != second:
            return True


valid = 0
for raw_line in open("./input").readlines():
    line = Line(raw_line)
    if line.check():
        valid += 1

print(valid)

File: day_two/part_two.rb
class Line
  def initialize(raw_line)
    @raw_line = raw_line
    separate_vals()
  end
  def separate_vals()
    nums, letter, @password = @raw_line.split(" ")
    @letter = letter[0]
    first, second = nums.split("-")
    @first = first.to_i
    @second = second.to_i
  end
  def check()
    if @password[@first-1] == @letter and @password[@second-1] != @letter
      return true
    elsif @password[@first-1] != @letter and @password[@second-1] == @letter
      return true
    else
      return false
    end
  end
end

valid = 0
File.readlines("./input").each do |raw_line|
  line = Line.new(raw_line)
  if line.check()
    valid += 1
  end
end
puts valid
